src/components/Form/elements/NumberInput.tsx

```diff
@@ -1,17 +1,115 @@
-import { getNums, parsePhoneNumber } from '../helpers';
-import Input, { TextInputProps } from './Input';
+import { kebabCase } from 'lodash';
+import { FC, useRef, useState } from 'react';
+import styled from 'styled-components';
+import ClearButton from './ClearButton';
+import {
+  TextInputContainerStyles,
+  TextInputLabelStyles,
+  TextInputProps,
+  TextInputStyles,
+} from './Input';
+import RequiredStatus from './RequiredStatus';
 
-// TODO: Rename to `PhoneInput`
-const NumberInput = ({ onChange, ...props }: TextInputProps) => {
+type NumberInputProps = {
+  min?: number;
+  max?: number;
+  step?: number;
+} & TextInputProps;
+
+const NumberInput: FC<NumberInputProps> = ({
+  onChange,
+  min,
+  max,
+  step,
+  type = 'text',
+  name,
+  value,
+  label,
+  isErr,
+  required,
+  placeholder,
+  isValid,
+  animation,
+  style,
+  testId,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isFocused, setIsFocus] = useState<boolean>(false);
+
+  const isValidMinMax = () => {
+    if (min !== undefined && max !== undefined) {
+      return parseFloat(value) >= min && parseFloat(value) <= max;
+    }
+    if (min !== undefined) {
+      return parseFloat(value) >= min;
+    }
+    if (max !== undefined) {
+      return parseFloat(value) <= max;
+    }
+    return true;
+  };
```

Comment:
Definitely a nit, but why not parse the float once at the top?
```suggestion
  const isValidMinMax = () => {
    const val = parseFloat(value);
    if (min !== undefined && max !== undefined) {
      return val >= min && val <= max;
    }
    if (min !== undefined) {
      return val >= min;
    }
    if (max !== undefined) {
      return val <= max;
    }
    return true;
  };

```

Reply:
because copilot wrote it and it was technically fine..

Reply:
parsed once https://github.com/OperationSpark/operationspark-org-website-2022/pull/206/commits/e9e6a48107ed0cc026e891ce195348342ad14eda